packages/ui/src/components/Button/Button.tsx

```diff
@@ -0,0 +1,78 @@
+import type { ButtonHTMLAttributes, PropsWithChildren } from 'react';
+import { forwardRef } from 'react';
+import { twMerge } from 'tailwind-merge';
+
+import { Icon } from './Icon';
+import type { BaseProps } from './types';
+
+export interface ButtonProps extends BaseProps, ButtonHTMLAttributes<HTMLButtonElement> {
+    isLoading?: boolean;
+    isDisabled?: boolean;
+    onClick?: () => void;
+    contentClassName?: string;
+}
+
+export const Button = forwardRef<
+    HTMLButtonElement,
+    Omit<PropsWithChildren<ButtonProps>, 'onResize' | 'onResizeCapture'>
+>(
+    (
+        {
+            variation = 'primary',
+            className,
+            type = 'button',
+            size = 'default',
+            icon,
+            iconPlacement = 'left',
+            isLoading,
+            isDisabled,
+            onClick,
+            children,
+            contentClassName,
+            ...buttonProps
+        },
+        ref,
+    ) => (
+        <button
+            ref={ref}
+            // eslint-disable-next-line react/button-has-type
+            type={type}
+            className={twMerge(
+                'rounded-[4px] flex items-center justify-center border-[1px] border-transparent bg-transparent',
```

Comment:
You're using the default Tailwind values, so no need for arbitrary syntax

```suggestion
                'rounded flex items-center justify-center border border-transparent bg-transparent',
```